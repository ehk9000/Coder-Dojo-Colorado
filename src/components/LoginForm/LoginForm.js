import React, { Component } from 'react';
import { Redirect, Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import { connect } from 'react-redux';
import * as actions from '../../actions';
import requests from '../../utils/requests/requests';

export class LoginForm extends Component {
  state = {
    username: '',
    password: '',
    success: false,
    role: null
  }

  handleChange = (e) => {
    const { name, value } = e.target;

    this.setState({
      [name]: value
    });
  }

  handleLogin = (e) => {
    const { username, password } = this.state;
    const user = {
      username,
      password
    };
    
    e.preventDefault();
    this.signIn(user);
  }

  signIn = async (user) => {
    const { addUser } = this.props;
    const result = await requests.signIn(user);

    if (result) {
      const { user: validUser, token } = result.signIn;
      
      localStorage.setItem('token', JSON.stringify(token));
      addUser(validUser);
      this.setState({ success: true, role: validUser.role });
    }
  }

  render() {
    const { success, role } = this.state;
    const { error, isFetching } = this.props;
    const roleAssign = () => {
      if (role === 0) {
        return '/dashboard/student';
      } else if (role === 1) {
        return '/myfamily';
      } 
      return '/dashboard/admin';
    };

    if (success) return <Redirect to={`${roleAssign()}`} />;

    return (
      <form
        className="LoginForm"
        onSubmit={this.handleLogin}
      >
        <h2>Login</h2>
        <label htmlFor="username-input">
          Username:
          <input
            id="username-input"
            type="text"
            name="username"
            onChange={this.handleChange}
          />
        </label>
        <label htmlFor="password-input">
          Password:
          <input
            id="password-input"
            type="password"
            name="password"
            onChange={this.handleChange}
          />
        </label>
        { error && <p className="error-msg">{ error }</p>}
        { isFetching && <p>Logging in, please wait...</p>}
        <button type="submit" className="signin-btn">LOGIN</button>
        <p className="register-link">
          Don&#39;t have an account?&nbsp;&nbsp;
          <Link to="/register"><span>Register Here</span></Link>
        </p>
      </form>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  addUser: user => dispatch(actions.addUser(user))
});

export const mapStateToProps = ({ isFetching, error }) => ({
  error,
  isFetching
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginForm);

LoginForm.propTypes = {
  addUser: PropTypes.func,
  error: PropTypes.string,
  isFetching: PropTypes.bool
};

LoginForm.defaultProps = {
  addUser: () => {},
  error: '',
  isFetching: false
};
