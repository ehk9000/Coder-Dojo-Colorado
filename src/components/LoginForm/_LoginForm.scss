.LoginForm {
  @include flex-column;
  margin: 10% auto;
  width: 500px;
  min-width: 300px;
  padding: 40px;
  background-color: #FFFFFF;

  h2 {
    margin: 0 0 15px 0;
    text-align: center;
  }

  label {
    @include flex-column;
    margin: 0 0 2px 0;
    color: #333532;
  }

  #username-input,
  #password-input {
    @include form-input-style
  }

  .signin-btn {
    @include form-submit;
    background-color: $orange;
  }

  .register-link {
    text-align: center;
    margin: 15px 15px 0 15px;

    span {
      color: #000000;
    }
  }
}