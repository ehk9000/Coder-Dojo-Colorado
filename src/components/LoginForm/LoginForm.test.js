import React from 'react';
import LoginForm from './LoginForm';
import { shallow } from 'enzyme';

describe('LoginForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(< LoginForm /> )
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})