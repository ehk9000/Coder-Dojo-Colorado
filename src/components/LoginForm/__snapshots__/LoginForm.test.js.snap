// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should match snapshot 1`] = `
<form
  className="LoginForm"
  onSubmit={[Function]}
>
  <h2>
    Login
  </h2>
  <label
    htmlFor="username-input"
  >
    Username:
    <input
      id="username-input"
      name="username"
      onChange={[Function]}
      type="text"
    />
  </label>
  <label
    htmlFor="password-input"
  >
    Password:
    <input
      id="password-input"
      name="password"
      onChange={[Function]}
      type="password"
    />
  </label>
  <div
    className="error-msg"
  />
  <button
    className="signin-btn"
    type="submit"
  >
    LOGIN
  </button>
  <p
    className="register-link"
  >
    Don't have an account?
    <Link
      to="/register"
    >
      <span>
         Register Here
      </span>
    </Link>
  </p>
</form>
`;
